base/loading.jl

```diff
@@ -1960,14 +1960,21 @@ const include_callbacks = Any[]
 const _concrete_dependencies = Pair{PkgId,UInt128}[] # these dependency versions are "set in stone", and the process should try to avoid invalidating them
 const _require_dependencies = Any[] # a list of (mod, abspath, fsize, hash, mtime) tuples that are the file dependencies of the module currently being precompiled
 const _track_dependencies = Ref(false) # set this to true to track the list of file dependencies
-function _include_dependency(mod::Module, _path::AbstractString; track_content=true)
+function _include_dependency(mod::Module, _path::AbstractString; track_content=true,
+                             path_maybe_dir=false)
     prev = source_path(nothing)
     if prev === nothing
         path = abspath(_path)
     else
         path = normpath(joinpath(dirname(prev), _path))
     end
-    if _track_dependencies[]
+    if !_track_dependencies[]
+        if !path_maybe_dir && !isfile(path)
+            throw(ArgumentError("including $(repr(path)): No such file"))
+        elseif path_maybe_dir && !ispath(path)
+            throw(ArgumentError("including $(repr(path)): No such file or directory"))
```

Comment:
These error types are wrong: it is not an argument error to pass in a valid string, but it is a SystemError for that not to be a path that can be opened by the user. You might even want to use the new `Filesystem.isreadable` here, since that is faster and may be more accurate here? Currently failing this test:
```
Test Failed at /cache/build/tester-amdci5-9/julialang/julia-master/julia-d8be815f40/share/julia/stdlib/v1.12/Distributed/test/distributed_exec.jl:1580
  Expression: exc isa SystemError
   Evaluated: ArgumentError("including \"/cache/build/tester-amdci5-9/julialang/julia-master/julia-d8be815f40/share/julia/stdlib/v1.12/Distributed/test/testfile\": No such file") isa SystemError
```

Reply:
There is a chicken-egg problem with fixing this test failure (which I already mentioned above): IIUC, then
- I need to PR `Distributed.jl` and add a `@static if VERSION == v"1.11-XXX"` to update the test there,
- but to get the commit nr `XXX` I need to first merge this PR.

So either we first merge this, then fix Distributed's tests and then bump the stdlib.
Or I just revert the `SystemError`'s prefix from "including" to "opening" and everything is fine.
I tend towards the latter, what do you say?

Reply:
I'd suggest: Do the latter now so that we can get this merged. And if someone is unhappy about the text of the error message ww can update that later in a separate PR.